src/app/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { useLocation } from 'react-router-dom';
+import useGroupDetails from '../../hooks/useGroupDetails';
+import Button from '../Button/Button';
+import CloseButton from '../CloseButton/CloseButton';
+import BoxingIcon from '../Icons/BoxingIcon';
+import Schedule from '../Schedule/Schedule';
+import styles from './Modal.module.css';
+
+type ModalProps = {
+  className?: string;
+  open: boolean;
+  onClose: () => void;
+};
+
+export default function Modal({ className, onClose }: ModalProps): JSX.Element {
+  const query = new URLSearchParams(useLocation().search);
+  const groupname = query.get('groupname');
+  const { groupDetails, isLoading, errorMessage } = useGroupDetails(groupname);
+
+  if (errorMessage) {
+    return <div>Error</div>;
+  }
+
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
+
+  if (!groupDetails) {
+    return <div>Group not found</div>;
+  }
+
+  return (
+    <div className={className}>
+      <div className={styles.card}>
+        <header className={styles.header}>
+          <div className={styles.closeButton}>
+            <CloseButton onClick={onClose} />
+          </div>
+          <div className={styles.joinButton}>
+            <Button variant="tertiary">Join</Button>
+          </div>
```

Comment:
are all these divs really needed? maybe you can give the styles to the button directly for example. it also looks like you have two divs around the whole component?